### ISSUE TITLE
C#: Can't compare ButtonIndex property (int) to ButtonList enum value, e.g. ButtonList.Left

### ISSUE THREAD

**Issue by MinimumEntropy** [godotengine/godot#41552]
**Godot version:** 3.2.2.stable.mono.official

**OS/device including version:** Windows 10

**Issue description:** 
I wrote **C#** code to catch the `_Input` event, then checked for multiple clicks by first casting to `InputEventMouseButton` with the `is` keyword, then checking if the `InputEventMouseButton`'s `ButtonIndex.Equals(ButtonList.Left)`. This never was true. However, upon replacing the `ButtonList.Left` with simply `1`, it works perfectly. I find this strange as it works with (for example) `WheelUp` and `WheelDown`.

**Steps to reproduce:** Please simply download below, run, and observe the debugging output! My code is also written below:

```cs
using Godot;

public class example : Spatial
{
    public override void _Input(InputEvent inputEvent)
    {
        if (inputEvent is InputEventMouseButton buttonEvent)
        {
            if (buttonEvent.IsPressed()) // limits prints per click to one for readability, can remove if needed
            {
                if (buttonEvent.ButtonIndex.Equals(1)) GD.Print("Mouse Button Pressed Using 1");
                if (buttonEvent.ButtonIndex.Equals(ButtonList.Left)) GD.Print("Mouse Button Pressed Using ButtonList");
            }
        }
    }
}
```

**Minimal reproduction project:** 
[MinimalExampleForButtonListIssue.zip](https://github.com/godotengine/godot/files/5133181/MinimalExampleForButtonListIssue.zip)

**Comment by akien-mga** [comment]
I'm not sure it's a bug, C# probably doesn't auto-cast enum types to int, and `buttonEvent.ButtonIndex` returns an int and not an enum value. Casting `ButtonList.Left` to `int` or `buttonEvent.ButtonIndex` to `ButtonList` works fine:
```csharp
                if (buttonEvent.ButtonIndex.Equals(1)) GD.Print("Mouse Button Pressed Using 1");
                if (((ButtonList)buttonEvent.ButtonIndex).Equals(ButtonList.Left)) GD.Print("Mouse Button Pressed Using ButtonList (cast as enum)");
                if (buttonEvent.ButtonIndex.Equals((int)ButtonList.Left)) GD.Print("Mouse Button Pressed Using ButtonList (cast to int)");
```

I agree that it's not the best UX wise, but I'm not sure how this can be solved in C#. CC @neikeq

**Comment by neikeq** [comment]
Can `ButtonIndex` return anything other than a value from the enum? If not then then it should return `ButtonList` instead of `int`.

**Comment by neikeq** [comment]
This was going to be fixed by #27450. It's pending being salvaged by @aaronfranke. Unless he can't work on it anymore, in which case I would take care of it at some point before 4.0.

**Comment by aaronfranke** [comment]
I can't salvage #27450 without conflicting with #38054, so I would prefer that be merged first.

**Comment by neikeq** [comment]
@aaronfranke Yeah, no hurries. 4.0 is still far.

**Comment by raulsntos** [comment]
This was implemented by #47378
